ext/sockets/sockets.c

```diff
@@ -2300,6 +2303,22 @@ PHP_FUNCTION(socket_set_option)
 		}
 #endif
 
+#if defined(UDP_SEGMENT)
```

Comment:
This option is only valid for level==SOL_UDP, right? If so, don't we need to check that in this case too?

Reply:
it is usually more relevant to checkwhen the value is supposed to be a non integral type ... then using an int by accident still set size type of the said struct. or vice versa.